memorymap.go

```diff
@@ -57,3 +57,17 @@ func (mm *memoryMap) iter() <-chan memoryMapItem {
 
 	return c
 }
+
+func (mm *memoryMap) copy() *memoryMap {
+	mm.Lock()
+	defer mm.Unlock()
+
+	m := make(map[string][]byte, len(mm.m))
+	for k, v := range mm.m {
+		m[k] = v
+	}
+
+	return &memoryMap{
+		m: m,
+	}
+}
```

Comment:
* Originally, I just wanted to increase the buffer on the iterator channel to minimize context-switching in the scheduler
	* basically, when the chan buffer is 0 and the size of the map is 1000, the scheduler will have to switch between goroutines 1000 times to read all the values
* But then I realized that the usage of the iter is only for reading the whole map, which can be done efficiently
	* Are there any other usage of the iter on your mind or only this? Because if only the whole map read - we can remove it.
* In the code above, the compiler does a few allocation optimizations during map copying 
* Iterating over the map and allocating `memoryMapItem` per each item can be avoided, so as goroutine